core/trino-main/src/main/java/io/trino/operator/output/PositionsAppender.java

```diff
@@ -39,8 +39,15 @@ public TypedPositionsAppender(Type type)
         public void appendTo(IntArrayList positions, Block source, BlockBuilder target)
         {
             int[] positionArray = positions.elements();
-            for (int i = 0; i < positions.size(); i++) {
-                type.appendTo(source, positionArray[i], target);
+            if (source.mayHaveNull()) {
+                for (int i = 0; i < positions.size(); i++) {
+                    type.appendTo(source, positionArray[i], target);
+                }
+            }
+            else {
```

Comment:
remove. I dont think it matters. Branch prediction would work the same

Reply:
removed (dedicated `PositionsAppenders` are anyway in cases where this could have impact)